src/GeneratorV2/Process/Overloaders.cs

```diff
@@ -169,6 +169,229 @@ public void WritePrologue(IndentedTextWriter writer, NameTable nameTable)
+                            Length = null
+                        };
+                        spanParameters[i] = parameter with
+                        {
+                            Type = new CSSpan(vector, ptr.Constant || baseType.Constant),
+                            Length = null
+                        };
+                        arrayParameters[i] = parameter with
+                        {
+                            Type = new CSArray(vector, ptr.Constant || baseType.Constant),
+                            Length = null
+                        };
+                        nameTable.Rename(parameter, parameter.Name + "_ptr");
+                        ptrParam = parameter;
+                        vectorParam = singleParameters[i];
+                    }
+                }
+            }
+
+            if (countParameter == null || ptrParam == null || vectorParam == null)
+            {
+                newOverloads = null;
+                return false;
+            }
+
+            string overloadName = overload.OverloadName.Remove(vectorMatch.Index + vectorMatch.Length - 1, 1);
+            newOverloads = new List<Overload>()
+            {
+                overload with
+                {
+                    OverloadName = overloadName,
+                    InputParameters = singleParameters.Where(p => p != countParameter).ToArray(),
+                    NameTable = nameTable,
+                    NestedOverload = overload,
+                    MarshalLayerToNested = new SingleVectorLayer(countParameter, ptrParam, vectorParam)
+                },
+                overload with
+                {
+                    OverloadName = overloadName,
+                    InputParameters = spanParameters,
+                    NameTable = nameTable,
+                    NestedOverload = overload,
+                    MarshalLayerToNested = new SpanArrayVectorLayer(countParameter, ptrParam, vectorParam)
+                },
+                overload with
+                {
+                    OverloadName = overloadName,
+                    InputParameters = arrayParameters,
+                    NameTable = nameTable,
+                    NestedOverload = overload,
+                    MarshalLayerToNested = new SpanArrayVectorLayer(countParameter, ptrParam, vectorParam)
+                },
+            };
+            return true;
+        }
+
+        public record SingleVectorLayer(
+            Parameter CountParameters, Parameter PtrParam, Parameter VectorParam) : IOverloadLayer
+        {
+            private CsScope _csScope;
+            public void WritePrologue(IndentedTextWriter writer, NameTable nameTable)
+            {
+                BaseCSType vectorType = ((CSRef)VectorParam.Type).ReferencedType;
+                writer.WriteLine($"{CountParameters.Type.ToCSString()} {nameTable[CountParameters]} = 1;");
+                writer.WriteLine($"fixed ({vectorType.ToCSString()}* tmp_vecPtr = &{nameTable[VectorParam]})");
```

Comment:
Having a hardcoded name `tmp_vecPtr` here is not ideal as this overloader would not be able to be run twice, but as we don't have that use case in our code atm it doesn't matter. It would only matter if we wanted to part this code to stuff like vulkan in the future, but what is work that can happen if/whenever that happens.